kiteconnect/connect.py

```diff
@@ -637,6 +650,102 @@ def trigger_range(self, transaction_type, *instruments):
+        """Fetch details of a GTT"""
+        return self._get("gtt.info", {"trigger_id": trigger_id})
+
+    def _get_gtt_payload(self, trigger_type, tradingsymbol, exchange, trigger_values, last_price, orders):
+        """Get GTT payload"""
+        # Validations.
+        assert trigger_type in [self.GTT_OCO, self.GTT_SINGLE]
+        if type(trigger_values) != list:
+            raise ex.InputException("invalid type for `trigger_values`")
+        if trigger_type == self.GTT_SINGLE and len(trigger_values) != 1:
+            raise ex.InputException("invalid `trigger_values` for single leg order type")
+        elif trigger_type == self.GTT_OCO and len(trigger_values) != 2:
+            raise ex.InputException("invalid `trigger_values` for two-leg order type")
+
+        condition = {
+            "exchange": exchange,
+            "tradingsymbol": tradingsymbol,
+            "trigger_values": trigger_values,
+            "last_price": last_price,
+        }
+
+        gtt_orders = []
+        for o in orders:
+            # Assert required keys inside gtt order.
+            for req in ["transaction_type", "quantity", "price"]:
+                if req not in o:
+                    raise ex.InputException("`{req}` missing inside orders".format(req=req))
+
+            gtt_orders.append({
+                "exchange": exchange,
+                "tradingsymbol": tradingsymbol,
+                "transaction_type": o["transaction_type"],
+                "quantity": int(o["quantity"]),
+                "order_type": self.ORDER_TYPE_LIMIT,
+                "product": self.PRODUCT_CNC,
+                "price": float(o["price"]),
+            })
+
+        return condition, gtt_orders
+
+    def place_gtt(
+        self, trigger_type, tradingsymbol, exchange, trigger_values, last_price, orders
+    ):
+        """
+        Place GTT order
+
+        - `trigger_type` The type of GTT order(single/two-leg).
+        - `tradingsymbol` Trading symbol of the instrument.
+        - `exchange` Name of the exchange.
+        - `trigger_values` Trigger values (json array).
+        - `last_price` Last price of the instrument at the time of order placement.
+        - `orders` JSON order array containing following fields
+            - `transaction_type` BUY or SELL
+            - `quantity` Quantity to transact
+            - `price` The min or max price to execute the order at (for LIMIT orders)
+        """
+        condition, gtt_orders = self._get_gtt_payload(trigger_type, tradingsymbol, exchange, trigger_values, last_price, orders)
+
+        return self._post("gtt.place", {
+            "condition": json.dumps(condition),
+            "orders": json.dumps(gtt_orders),
+            "type": trigger_type})
+
+    def modify_gtt(
+        self, trigger_id, trigger_type, tradingsymbol, exchange, trigger_values, last_price, orders
+    ):
+        """
+        Modify GTT order
+
+        - `trigger_type` The type of GTT order(single/two-leg).
```

Comment:
Remove `trigger_type`, `tradingsymbol` and `exchange` from params since these cannot be modified. You need to refactor `_get_gtt_payload` method also to validate based on place or modify.